samples/subsys/mgmt/mcumgr/smp_svr/sample.mcumg.smp_svr.bt fails to build

When I run:

`./scripts/sanitycheck -p nrf52840dk_nrf52840 -s samples/subsys/mgmt/mcumgr/smp_svr/sample.mcumg.smp_svr.bt`

I get:

```
[ 77%] Building C object zephyr/subsys/mgmt/mcumgr/CMakeFiles/subsys__mgmt__mcumgr.dir/buf.c.obj
[ 77%] Building C object zephyr/subsys/mgmt/mcumgr/CMakeFiles/subsys__mgmt__mcumgr.dir/smp.c.obj
[ 78%] Building C object zephyr/subsys/mgmt/mcumgr/CMakeFiles/subsys__mgmt__mcumgr.dir/smp_bt.c.obj
[ 78%] Building C object zephyr/subsys/mgmt/mcumgr/CMakeFiles/subsys__mgmt__mcumgr.dir/smp_shell.c.obj
/home/galak/git/zephyr/subsys/mgmt/mcumgr/smp_shell.c: In function 'smp_shell_tx_raw':
/home/galak/git/zephyr/subsys/mgmt/mcumgr/smp_shell.c:145:38: error: initialization of 'const struct shell_uart * const' from incompatible pointer type 'const struct shell *' [-Werror=incompatible-pointer-types]
  145 |  const struct shell_uart *const su = shell_backend_uart_get_ptr();
      |                                      ^~~~~~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
gmake[2]: *** [zephyr/subsys/mgmt/mcumgr/CMakeFiles/subsys__mgmt__mcumgr.dir/build.make:122: zephyr/subsys/mgmt/mcumgr/CMakeFiles/subsys__mgmt__mcumgr.dir/smp_shell.c.obj] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:3448: zephyr/subsys/mgmt/mcumgr/CMakeFiles/subsys__mgmt__mcumgr.dir/all] Error 2
gmake: *** [Makefile:104: all] Error 2
```

--- a/subsys/mgmt/mcumgr/smp_shell.c
+++ b/subsys/mgmt/mcumgr/smp_shell.c
@@ -149,7 +149,8 @@ static uint16_t smp_shell_get_mtu(const struct net_buf *nb)
 
 static int smp_shell_tx_raw(const void *data, int len, void *arg)
 {
-	const struct shell_uart *const su = shell_backend_uart_get_ptr();
+	const struct shell *const sh = shell_backend_uart_get_ptr();
+	const struct shell_uart *const su = sh->iface->ctx;
 	const struct shell_uart_ctrl_blk *const scb = su->ctrl_blk;
 	const uint8_t *out = data;
 

